hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -235,7 +235,13 @@ object Simplify {
 
       case TableMapGlobals(child, Ref("global", _)) => child
 
-      case MatrixMapRows(child, Ref("va", _), None) => child
+      case MatrixMapRows(child, Ref("va", _)) => child
+
+      case t@MatrixKeyRowsBy(MatrixKeyRowsBy(child, _, _), keys, false) if canRepartition(t) =>
+        MatrixKeyRowsBy(child, keys, false)
+
+      case t@MatrixKeyRowsBy(MatrixKeyRowsBy(child, _, true), keys, true) if canRepartition(t) =>
```

Comment:
Why does this only work if the inner KeyRowsBy is `true` where as the one above works in both cases?

Reply:
In the `MatrixKeyRowsBy(MatrixKeyRowsBy(child, keys1, false), keys2, true)` case, we're first shuffling to `keys1`, then saying "change the keys to `keys2`, but I know it's already sorted". That certainly can't simplify to a single `MatrixKeyRowsBy` with `isSorted = true`, because we don't know anything about the order before the first shuffle. It's a weird case. I think the only way it could be valid is if `keys2` is a prefix of `keys1`, and it almost looks like you could simplify that to `MatrixKeyRowsBy(child, keys2, false)`, but that isn't exactly the same: both forms end up keyed by `keys2`, but the original is also sorted by the longer `keys1`, which isn't true in the simplified form.

Reply:
ok.